### PR TITLE
Add names to retroactive attendance form

### PR DESCRIPTION
**Pull request by ryanDing26**
# Info

Closes **#322**. 

# Description

What changes did you make? List all distinct problems that this PR addresses. Explain any relevant
motivation or context.

Faris wanted to change the functionality on the retroactive attendance form (within the admin dashboard) so that it displayed names in the format `firstName lastName (email)` so that it would be easier to autocomplete manual attendance by name instead of asking people for their email for points.

## Changes

- Changed the emailRaw variable in `getAllEmails` in UserRepository.ts such that it built a query with the firstName and lastName of a user from the UserModel object alongside the existing email selection
- Changed the return statement of `getAllEmails` in UserRepository.ts such that instead of just mapping all the emails, it would map `firstName lastName (email)`

# Type of Change

- [ ] Patch (non-breaking change/bugfix)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Documentation (A change to a README/description)
- [ ] Continuous Integration/DevOps Change (Related to deployment steps, continuous integration
      workflows, linting, etc.)
- [ ] Other: (Fill In) 

If you've selected Patch, Minor, or Major as your change type, **make sure to bump the version before merging in `package.json`!**
# Testing

I have tested that my changes fully resolve the linked issue ...

- [ ] locally.
- [ ] on the testing API/testing database.
- [ ] with appropriate Postman routes. Screenshots are included below.

# Checklist

- [ ] I have performed a self-review of my own code.
- [ ] I have followed the style guidelines of this project.
- [ ] I have appropriately edited the API version in the `package.json` file.
- [x] My changes produce no new warnings.

# Screenshots

Please include a screenshot of your Postman testing passing successfully.
image

### CODE REVIEW

File: repositories/UserRepository.ts [hunk 1]
```diff
@@ -48,9 +48,9 @@ export class UserRepository extends BaseRepository<UserModel> {
   public async getAllEmails(): Promise<string[]> {
     const emailsRaw = await this.repository
```
Comment (nik-dange) [inline repositories/UserRepository.ts:49]: can we change this variable name to `emailsNamesRaw` so it's a bit more descriptive?

**Review by nik-dange — CHANGES_REQUESTED** [review]
Just a small nit. You'll need to fix your tests and run the linter before being able to merge though, make sure those pass before opening a PR

**Comment by dowhep** [comment]
Nit: Since you will be returning an array of strings that concatenate names and emails, it would be cleaner and more correct to define a new type for such strings. Maybe make the field name "namesAndEmails" or something.

**Review by nik-dange — APPROVED** [review]
lgtm. **please bump package version before merging**

**Comment by ryanDing26** [comment]
This is the final version I would like to submit for the PR. Let me know if any more changes need to be made!

File: types/ApiResponses.ts [hunk 2]
```diff
@@ -299,6 +298,11 @@ export interface FulfillMerchOrderResponse extends ApiResponse {
 }
 
 // USER
+export interface NameEmail {
+  firstName: string;
+  lastName: string;
+  email: string;
+}
```
Comment (nik-dange) [inline types/ApiResponses.ts:305]: I think this type might be better under `types/internal`, thoughts?

File: services/UserAccountService.ts [hunk 3]
```diff
@@ -167,10 +167,10 @@ export default class UserAccountService {
     });
   }
 
-  public async getAllEmails(): Promise<string[]> {
+  public async getAllNamesEmails() {
```
Comment (nik-dange) [inline services/UserAccountService.ts:170]: nit: rename to getAllNamesAndEmails

File: repositories/UserRepository.ts [hunk 4]
```diff
@@ -45,12 +45,16 @@ export class UserRepository extends BaseRepository<UserModel> {
     return this.repository.findOne({ accessCode });
   }
 
-  public async getAllEmails(): Promise<string[]> {
-    const emailsRaw = await this.repository
+  public async getAllNamesEmails(): Promise<NameEmail[]> {
```
Comment (nik-dange) [inline repositories/UserRepository.ts:48]: nit: rename to getAllNamesAndEmails

**Review by nik-dange — CHANGES_REQUESTED** [review]
One quick question

File: types/ApiResponses.ts (hunk 2 above)
Comment (nik-dange) [reply types/ApiResponses.ts:305]: nvm, seems fine

File: types/ApiResponses.ts [hunk 5]
```diff
@@ -299,6 +298,11 @@ export interface FulfillMerchOrderResponse extends ApiResponse {
 }
 
 // USER
+export interface NameEmail {
```
Comment (nik-dange) [inline types/ApiResponses.ts:301]: nit: rename to NameAndEmail

File: services/UserAccountService.ts [hunk 6]
```diff
@@ -167,10 +167,10 @@ export default class UserAccountService {
     });
   }
 
-  public async getAllEmails(): Promise<string[]> {
+  public async getAllNamesAndEmails() {
```
Comment (echang594) [inline services/UserAccountService.ts:170]: Can you add the return type back

**Review by echang594 — DISMISSED** [review]
See my other comment

**Review by nik-dange — APPROVED** [review]
@ryanDing26 looks good to me, just threw in a return type and did some listing

**Pull request merged**